xdsl/dialects/stencil.py

```diff
@@ -289,6 +290,50 @@ def __init__(self, float_t: AnyFloat) -> None:
         super().__init__([float_t])
 
 
+@irdl_op_definition
+class ApplyOp(IRDLOperation):
+    """
+    This operation takes a stencil function plus parameters and applies
+    the stencil function to the output temp.
+
+    Example:
+
+      %0 = stencil.apply (%arg0=%0 : !stencil.temp<?x?x?xf64>) -> !stencil.temp<?x?x?xf64> {
+        ...
+      }
+    """
+
+    name = "stencil.apply"
+    args: Annotated[VarOperand, Attribute]
+    region: Region
+    res: Annotated[VarOpResult, TempType]
+
+    @staticmethod
+    def get(
+        args: Sequence[SSAValue] | Sequence[Operation],
+        body: Block,
+        result_types: Sequence[TempType[Attribute]],
+    ):
+        assert len(result_types) > 0
+
+        return ApplyOp.build(
+            operands=[list(args)],
+            regions=[Region(body)],
+            result_types=[result_types],
+        )
+
+    def verify_(self) -> None:
+        if len(self.res) < 1:
+            raise VerifyException(
+                f"Expected stencil.apply to have at least 1 result, got {len(self.res)}"
+            )
```

Comment:
Shouldn't you also verify that the input and block argument types match, for some definition of match?

Reply:
Indeed, this PR is non exhaustive :wink:

Reply:
Well, then we should be a bit careful with moving it out of experimental maybe? idk.

Reply:
It was all discussed there and done in a separate PR:
https://xdsl.zulipchat.com/#narrow/stream/362011-xDSL/topic/stencil.20dialect/near/364550623

Reply:
There's a lot of debating those things overall, and this is just a PR, about making the dialect a bit better supported.
I think there's a lot of energy put in those general discussions that could be focused on just the next step to get it polished enough TBH! This PR is just a step, not a big claim of what not :slightly_smiling_face:

Reply:
Marking as resolved as it does not concern the content of this PR.
Those missing bits are indeed to be added.